website/addons/base/views.py

```diff
@@ -548,19 +585,21 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
 
     if version is None:
         if file_node.get_guid():
-            # If this file has been successfully view before but no longer exists
-
-            # Move file to trashed file node
+            # If this file has been successfully view before, or OSF, but no
```

Comment:
I've made the comment more succinct. What I was trying to get at is that OSF storage files get GUIDs whether they are viewed or not.